### PR TITLE
ENH: multivariate mean tests and confint

### PR DESCRIPTION
**Pull request by josef-pkt**
see e.g. #3208

I thought I already wrote something like this but don't find anything.
This is currently just a short exercise writing some basic multivariate mean tests to get started.

plans for implementation

There is a lot of repeated computation if we provide and use separate function that all start from data. Those standalone functions are good to get started but need better structure. 
Most likely the pattern will or should follow weightstats with a mixture of classes and standalone functions. One useful pattern is to have the functions based on summary statistics, so we have to compute those only once.

Reference for this time is Johnson and Wichern Applied Multivariate Statistical Analysis 6th edition. (I started to browse it by chance.)

Stata mvtests has good functions with almost complete formulas. However, I didn't see anything about mv confidence intervals. The simulataneous confidence intervals here use an example from Johnson and Wickern. They look similar to or the same as the Sheffe intervals added by Kerby.
Pointwise confidence intervals are compared to weightstats.

Stata doesn't have any kurtosis correction in mvtests, AFAICS. (see elliptical issues)

### CODE REVIEW

**Comment by josef-pkt** [comment]
I guess there is no `__future__` in python 2.7, Travis fails with it.
(One more year before saying byebye, see numpy mailing list)

**Comment by josef-pkt** [comment]
a comment about the `df` in the F test, and TODO in unit tests:
Stata uses the number of linear combinations as df in the test of linear hypothesis.
I'm basing df on k_vars.
I'm not sure this is correct, but there is a difference between Sheffe type (e.g. of confidence intervals) which maintain the size for any linear combination, while the standard F/Wald test is for the specific joint hypothesis.

that is, we have (e.g. for confidence intervals):
(i) Sheffe: maintain size for all possible tests
(ii) subset: maintain size for a specific set of tests
(iii) individual/point wise: maintain size for a single non-composite hypothesis

my guess is that Stata does (ii) while I do (i) following Johnson and Wichern, but I might have it still wrong. (I didn't try to understand the math for the different cases. The second point is that we also vectorize individual tests as in t_test, which are not joint tests but for simultaneous confint we want to correct for multiple testing.)
#2955 #790 #2172 related on simultaneous confidence intervals and Sheffe

**Comment by jbrockmendel** [comment]
@josef-pkt I'll be happy to look this over if review will be useful.  Kinda at a loss for how to contribute these days.

**Comment by josef-pkt** [comment]
@jbrockmendel  It's a bit early to review this.

We need functionality like this and similar. But the occasion was writing code that is relatively limited and can be done in a few hours when I have time. I was only reading and opening issues in the last round in this area, at least I don't find any code.

**Comment by josef-pkt** [comment]
amended with division from the future

**Comment by josef-pkt** [comment]
will need rebase when working on this again
 (appveyor fails on unrelated compat issue that is fixed in master)

File: statsmodels/stats/multivariate.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    -------
+    low : ndarray
+        lower confidence bound on the linear transformed
+    upp : ndarray
+        upper confidence bound on the linear transformed
+    values : ndarray
+        mean or their linear transformation, center of the confidence region
+
+
+    Result 5.3 page 225
```
Comment (josef-pkt) [inline statsmodels/stats/multivariate.py]: What is the reference here?
no reference to a book or article in code/docstrings

**Comment by josef-pkt** [comment]
These look like nice simple functions that we should be able to merge independent of more elaborate enhancements.
unit test coverage looks ok based on a brief skimming of those.

File: statsmodels/stats/multivariate.py (hunk 1 above)
Comment (josef-pkt) [reply statsmodels/stats/multivariate.py:88]: Johnson and Wichern 6th edition, mentioned in the PR description

**Comment by josef-pkt** [comment]
It looks like simultaneous confint could be used for prediction in models based on params and cov_params,  `mv_mean_conf_int_simult_stat`
(names have too many underlines/words)

**Comment by josef-pkt** [comment]
rebased

**Comment by josef-pkt** [comment]
no failures besides lint
`mv_mean_conf_int_simult` has no unit test coverage

**Comment by josef-pkt** [comment]
this PR only includes 1-sample test and confidence interval

**Comment by josef-pkt** [comment]
possible merge the two confint function and add keyword `simult` or `simultaneous`
`simultaneous` is more difficult to spell

test_mvmean_1samp
confint_mvmean_1samp
question: onesided alternative, Stata doesn't have it but I guess we could add it with elementwise inequality (maybe that could become similar to oneway max diff equivalence testing (not Included in anova PR))

and to add
test_mvmean_ksamp or better oneway_mvmean (similar to anova/robust stats PR)
(I guess there could be also the equivalence test similar to oneway anova)

maybe (Stata doesn't have it as special case)
test_mv_mean_2samp


Stata has several null hypothesis in one sample case:

equal test that variables in varlist have equal means; the default
zero test that means of varlist are all equal to 0
equals(M) test that mean vector equals vector M
linear(V) test that mean vector of varlist satisfies linear hypothesis described by matrix V


aside: R package https://cran.r-project.org/web/packages/ICSNP/index.html
has rank based function
what about norm.ppf transform? (currently used as one of the cov estimators in robust PR)
Can we use marginal transformation? 

(and then we get into copulas, ....)

**Comment by josef-pkt** [comment]
I amended commit (only one so far) with pep-8 cleaning and removing `__future__` import

**Comment by josef-pkt** [comment]
merge when green, extensions mv mean testing left for another time

**Comment by josef-pkt** [comment]
lint check fails in statespace in master, not this PR

mergin

**Pull request merged**